### PR TITLE
Fix trust remote code

### PR DESCRIPTION
**Pull request by koukandre**
1. fix the issue with trust remote code in order to load jina-models, we should add the "hf_trust_remote_code": true variable in txt_cfg in the model_config.json file in order to enable it.
2. log only the specific metrics.
3. Do evaluation on the benchmarks at the beginning of the training.
4. Increase the timeout time to block the distributed process to die from timeout during evaluation.

### CODE REVIEW

**Review by bwanglzu — APPROVED** [review]
LGTM!

File: src/open_clip/hf_model.py [hunk 1]
```diff
@@ -117,15 +118,15 @@ def __init__(
         if transformers is None:
             raise RuntimeError("Please `pip install transformers` to use pre-trained HuggingFace models")
         if config is None:
-            self.config = AutoConfig.from_pretrained(model_name_or_path)
+            self.config = AutoConfig.from_pretrained(model_name_or_path,trust_remote_code=trust_remote_code)
```
Comment (gmastrapas) [inline src/open_clip/hf_model.py]:
```suggestion
            self.config = AutoConfig.from_pretrained(model_name_or_path, trust_remote_code=trust_remote_code)
```

File: src/training/evaluate.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 from .distributed import is_master
 from .precision import get_autocast
 
+MTEB_LOGGING_METRICS = ['ncdg_at_10', 'cos_sim']
```
Comment (gmastrapas) [inline src/training/evaluate.py]: you mean ndcg?

File: src/training/main.py [hunk 3]
```diff
@@ -483,6 +474,20 @@ def main(args):
                 torch.save(checkpoint_dict, tmp_save_path)
                 os.replace(tmp_save_path, latest_save_path)
 
+        if (
+            any(v in data for v in ('val', 'imagenet-val', 'imagenet-v2'))
+            or args.clip_benchmark_frequency != 0
```
Comment (gmastrapas) [inline src/training/main.py:479]:
```suggestion
            or args.clip_benchmark_frequency != 0
            or args.mteb_frequency != 0
```

File: src/training/params.py [hunk 4]
```diff
@@ -168,6 +168,12 @@ def parse_args(args):
     parser.add_argument(
         "--save-frequency", type=int, default=1, help="How often to save checkpoints."
     )
+    parser.add_argument(
+        "--evaluate-on-start",
+        action="store_true",
+        default=False,
+        help="Do the first evaluation on the benchmarks before start training"
```
Comment (gmastrapas) [inline src/training/params.py]:
```suggestion
        help="Run the first benchmark evaluation before training"
```

**Review by gmastrapas — CHANGES_REQUESTED** [review]

**Review by gmastrapas — APPROVED** [review]

**Pull request merged**